crates/snforge-scarb-plugin/src/parse.rs

```diff
@@ -0,0 +1,66 @@
+use crate::attributes::{AttributeInfo, ErrorExt};
+use cairo_lang_diagnostics::DiagnosticsBuilder;
+use cairo_lang_filesystem::ids::{FileKind, FileLongId, VirtualFile};
+use cairo_lang_macro::Diagnostic;
+use cairo_lang_parser::{parser::Parser, utils::SimpleParserDatabase};
+use cairo_lang_syntax::node::{
+    ast::{FunctionWithBody, ModuleItem, OptionArgListParenthesized},
+    db::SyntaxGroup,
+    helpers::QueryAttrs,
+};
+use cairo_lang_utils::Upcast;
+use indoc::formatdoc;
+use std::sync::Arc;
+
+pub fn parse<T: AttributeInfo>(
```

Comment:
On Cairo main you can now use `db.parse_virtual(content)` instead of creating the virtual file yourself

Reply:
#2357